crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -348,24 +348,21 @@ impl<'db> UseDefMap<'db> {
         self.bindings_iterator(&self.bindings_by_use[use_id])
     }
 
-    /// Returns true if a given 'use' of a symbol is reachable from the start of the scope.
-    /// For example, in the following code, use `2` is reachable, but `1` and `3` are not:
-    /// ```py
-    /// def f():
-    ///     x = 1
-    ///     if False:
-    ///         x  # 1
-    ///     x  # 2
-    ///     return
-    ///     x  # 3
-    /// ```
-    pub(crate) fn is_symbol_use_reachable(&self, db: &dyn crate::Db, use_id: ScopedUseId) -> bool {
+    pub(super) fn is_reachable(
+        &self,
+        db: &dyn crate::Db,
+        reachability: ScopedVisibilityConstraintId,
+    ) -> bool {
         !self
             .visibility_constraints
-            .evaluate(db, &self.predicates, self.reachability_by_use[use_id])
+            .evaluate(db, &self.predicates, reachability)
             .is_always_false()
     }
 
+    pub(super) fn is_symbol_use_reachable(&self, db: &dyn crate::Db, use_id: ScopedUseId) -> bool {
+        self.is_reachable(db, self.reachability_by_use[use_id])
+    }
```

Comment:
Maybe it's worth adding a doc-comment here to note that this doesn't give you the "full" picture, because it's also important to consider whether the enclosing scope is reachable (and that `SemanticIndex::is_symbol_use_reachable` puts the two pieces of the puzzle together to give you the full picture)?

Reply:
That's a good suggestion. I'll add this in a follow up I'm already working on, where this function will change anyway.